client/executor/src/native_executor.rs

```diff
@@ -274,10 +274,20 @@ impl<D: NativeExecutionDispatch> NativeExecutor<D> {
 		default_heap_pages: Option<u64>,
 		max_runtime_instances: usize,
 	) -> Self {
-		let mut host_functions = D::ExtendHostFunctions::host_functions();
+		let extended =  D::ExtendHostFunctions::host_functions();
+		let mut host_functions = sp_io::SubstrateHostFunctions::host_functions()
```

Comment:
not sure about this. You are changing `host_functions `. Will this have no side effect?

Reply:
it just prioritizes user host functions over substrate's host functions if they have the same name, it's how the signature verification is bypassed

Reply:
thanks, sensible; someone with knowledge of this code should verify that it does not have any unintended side effects or foot guns.